### PR TITLE
Add torch.tensor.unique function

### PR DESCRIPTION
**Pull request by paulaehab**

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 1]
```diff
@@ -8237,3 +8237,47 @@ def test_torch_instance_multiply(
...
+
+
+# TODO: need to fix this test after fixing the tests reults from torch.unique
+# unique
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="torch.tensor",
+    method_name="unique",
+    dtype_x_axis=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("float"),
```
Comment (CatB1t) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:8249]: Ideally, we should be testing all `valid` dtypes.

**Review by CatB1t — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 2]
```diff
@@ -8237,3 +8237,47 @@ def test_torch_instance_multiply(
...
+
+    helpers.test_frontend_method(
+        init_input_dtypes=input_dtypes,
+        init_all_as_kwargs_np={"data": x[0]},
+        method_input_dtypes=input_dtypes,
+        method_all_as_kwargs_np={
+            "sorted": True,
+            "return_inverse": return_inverse,
+            "return_counts": return_counts,
+            "dim": axis,
```
Comment (CatB1t) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:8276]: We should also test the possibility of the axis of being `None`

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 3]
```diff
@@ -8237,3 +8237,47 @@ def test_torch_instance_multiply(
...
+    return_counts,
+):
+    input_dtypes, x, axis = dtype_x_axis
+
+    helpers.test_frontend_method(
+        init_input_dtypes=input_dtypes,
+        init_all_as_kwargs_np={"data": x[0]},
+        method_input_dtypes=input_dtypes,
+        method_all_as_kwargs_np={
+            "sorted": True,
```
Comment (CatB1t) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:8273]: we should generate a value for `sorted` too.

**Review by CatB1t — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py (hunk 2 above)
Comment (paulaehab) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:8276]: @CatB1t I have no idea how to do this, also I searched for all other function that accept axis= None all other members do the same so if you can give ma a hint I will be thankful .

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 4]
```diff
@@ -8237,3 +8237,49 @@ def test_torch_instance_multiply(
...
+# unique
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="torch.tensor",
+    method_name="unique",
+    dtype_x_axis=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        force_int_axis=True,
+        valid_axis=True,
+        min_num_dims=1,
```
Comment (CatB1t) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:8252]: We shouldn't limit the dimension size, `torch.unique` works fine with 0-dim tensors.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py (hunk 2 above)
Comment (CatB1t) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:8278]: We can have a composite strategy which either returns `None` or axis value, something like `draw(st.sample_from(axis, None))`

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
